interface/main/mobile/camera.php

```diff
@@ -0,0 +1,619 @@
+        }
+        ?>
+    ];
+    
+    
+    var reply = [];
+    <?php
+    if (!empty($setting_mRoom)) {
+        echo "var mRoom = ".text($setting_mRoom).";";
+    } else {
+        echo "var mRoom;";
+    }
+    ?>
+</script>
+
+<body style="background-color: #fff;" >
+<?php common_header($display); ?>
+
+<div id="gb-main" class="container-fluid">
+    <form id="save_media" name="save_media" action="#" method="post" enctype="multipart/form-data">
+
+        <div class="row">
+            <div class="col-xs-12 col-sm-12 col-md-12 col-lg-12 text-center">
+                <img src="<?php echo $GLOBALS['webroot']; ?>/interface/main/mobile/photobooth.png" id="head_img" alt="<?php echo xla('File Uploader'); ?>">
+            </div>
+            <div class="col-xs-12 col-sm-4 col-md-4 col-lg-4 text-center">
+                <div class="row text-center">
+                    <div class="col-xs-12 col-sm-12 col-md-12 col-lg-12 custom-file-upload">
+                        <span class="section_title"><?php echo xlt('Select Patient'); ?></span>
+                        <br />
+                        <div class="btn-group" data-toggle="buttons">
+                            <button class="btn btn-primary" id="byRoom"><?php echo xlt('By Room'); ?></button>
+                            <button class="btn btn-primary" id="byName"><?php echo xlt('By Name'); ?></button>
+                        </div>
+                        <br /><br />
+                        <select id="findRoom" name="findRoom" type="text"
+                                class="form-control byNameDisplay">
+                            <?php
+                                $rows = sqlStatement("SELECT * FROM list_options WHERE " .
+                                    "list_id = ? AND activity = 1 ", array('patient_flow_board_rooms'));
+                                while ($row = sqlFetchArray($rows)) {
+                                    $selected = ( ($row['option_id'] == $setting_mRoom) ? ' selected' : '');
+                                    echo "<option value='" . attr($row['option_id']) ."' ".$selected.">". text($row['title']) . "</option>\n";
+                                }
+                            ?></select>
+                        <?php
+                            /**
+                             * Logic to select patient who owns this data we are going to upload
+                             */
+                            
+                        if (!empty($results_byRoom)) {
+                            //we have a preference to select patient by Room, and we have a room, and we know who is in it!
+                            // $results_byRoom is the array holding these answers...
+                            $size = sqlNumRows($results_byRoom);
+                            if ($size == '1') {
+                                $row = sqlFetchArray($results_byRoom);
+                                $occupant = text($row['fname'] . " " . $row['lname']);
```

Comment:
generally better to escape when you output it rather than store in variables. Otherwise it's tough for developers to ensure stuff gets escaped. And good chance $occupant may get double escaped below at some point in the future.

Reply:
Also, note that some places below that use $occupant need to use attr() rather than text()